src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/StaTaskScheduler.cs

```diff
@@ -13,6 +13,9 @@ namespace Microsoft.Identity.Client.Platforms.Features.DesktopOs
     // This IDisposable class doe not need to implement Dispose method in standard way, because it is sealed.
     // If it ever needs to become inheritable, it should follow the standard pattern as desribed in http://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx.
     /// <summary>Provides a scheduler that uses STA threads.</summary>
+#if NET5_WIN
+    [System.Runtime.Versioning.SupportedOSPlatform("windows")]
```

Comment:
Why do we need to specify this? Is the default support 17663+ because of our target?

My understanding if I run MSAL on Win 7, then all these classes are available, unless they are marked with `SupportedOSPlatform("windows10.0.17763.0")`, which makes them available on 17763+.

Or is this to specify that this is available only on windows and not linux and such?

Reply:
It looks like 

`thread.SetApartmentState(ApartmentState.STA);`

is available only on Windows. 

I don't know why the analyzer is complaining about this, since net5-windows10.xyz is only available on Windows :) 
I'll log another bug on them I think.